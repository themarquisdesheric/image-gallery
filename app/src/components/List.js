import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DeleteButton from './DeleteButton';

class List extends Component {
  static propTypes = { 
    data: PropTypes.array.isRequired,
    onDelete: PropTypes.func.isRequired
  }

  render() {
    const { data, onDelete } = this.props;

    return (
      <div>
        {data.map(img => (
          <ul key={img._id}>
            <li className="list-title">
              {img.title}
            </li>
            <li>
              <a href="">
                {img.url}
              </a>
            </li>
            <li>
              {img.description}
            </li>
            <li>
              <DeleteButton onClick={() => onDelete(img._id)} />
            </li>
          </ul>
        ))}
      </div>
    );
  }
}

export default List;