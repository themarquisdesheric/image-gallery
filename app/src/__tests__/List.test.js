import React from 'react';
import renderer from 'react-test-renderer';
import shortid from 'shortid';
import List from '../components/List';

function onDelete() {}

const images = [
  {
    title: 'Black & Tan',
    description: 'Seriously heart-melting black and tan bunny... OMG <3',
    url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg',
    _id: shortid.generate()
  }
]

describe('List', () => {
  it('renders a list', () => {
    const tree = renderer.create(<List data={images} onDelete={onDelete} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});