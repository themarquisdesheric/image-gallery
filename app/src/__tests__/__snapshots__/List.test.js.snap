// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders a list 1`] = `
<div>
  <ul>
    <li
      className="list-title"
    >
      Black & Tan
    </li>
    <li>
      <a
        href=""
      >
        http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg
      </a>
    </li>
    <li>
      Seriously heart-melting black and tan bunny... OMG &lt;3
    </li>
    <li>
      <button
        className="delete-button"
        onClick={[Function]}
      >
        Delete ðŸ—‘
      </button>
    </li>
  </ul>
</div>
`;
